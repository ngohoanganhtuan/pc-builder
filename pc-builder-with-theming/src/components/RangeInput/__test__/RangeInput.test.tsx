import { render, fireEvent } from '@testing-library/react';
import * as renderer from 'react-test-renderer';

// Components
import { RangeInput } from '@components';

describe('Test [RangeInput] component', () => {
  const handleSetMin = jest.fn();
  const handleSetMax = jest.fn();
  const initialProps = {
    minValue: 0,
    maxValue: 100,
    handleMinChanged: handleSetMin,
    handleMaxChanged: handleSetMax,
  };

  afterEach(() => {
    handleSetMin.mockClear();
    handleSetMax.mockClear();
  });

  test('Component [RangeInput] should render correctly', () => {
    const component = renderer.create(<RangeInput {...initialProps} />).toJSON();
    expect(component).toMatchSnapshot();
  });

  test('Component [RangeInput] calls handle set min function when user enter values', () => {
    const { getByTestId } = render(<RangeInput {...initialProps} />);
    fireEvent.change(getByTestId('min-range-input'), {
      target: { value: '500' },
    });
    expect(handleSetMin).toHaveBeenCalled();
  });

  test('Component [RangeInput] calls handle set max function when user enter values', () => {
    const { getByTestId } = render(<RangeInput {...initialProps} />);
    fireEvent.change(getByTestId('max-range-input'), {
      target: { value: '1000' },
    });
    expect(handleSetMax).toHaveBeenCalled();
  });
});
