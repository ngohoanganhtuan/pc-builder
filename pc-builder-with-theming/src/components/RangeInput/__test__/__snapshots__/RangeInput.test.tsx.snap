// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test [RangeInput] component Component [RangeInput] should render correctly 1`] = `
<div
  className="range-slider d-flex flex-row justify-content-between align-items-center gap-2"
>
  <input
    className="form-control"
    data-testid="min-range-input"
    onChange={[MockFunction]}
    placeholder="Minimum"
    value={0}
  />
  ~
  <input
    className="form-control"
    data-testid="max-range-input"
    onChange={[MockFunction]}
    placeholder="Maximum"
    value={100}
  />
</div>
`;
